### ISSUE TITLE
dos-an-infinite-loop-ngiflib-c-669

### ISSUE THREAD

**Issue by Edward-L** [miniupnp/ngiflib#7]
in LoadGif ngiflig.c:669 has an infinite loop
if id dont match any case ,it wont break

poc:
[poc](https://github.com/Edward-L/fuzzing-pocs/blob/master/ngiflib/dos-an-infinite-loop-ngiflib-c-669.poc)

**Comment by Edward-L** [comment]
I think in GetByte ngiflib.c:73 have some problem. 

this demo also have an infinite loop, it seems likes yours
```
int main()
{
     int i=0;	
     FILE* fp=fopen("1.txt","r");
    while(i = getc(fp)){	
	printf("%c",i);
    }
    return 0;
}
```

**Comment by miniupnp** [comment]
I think it is fixed by 2cecb536e9a4adf1d4b48579f21fffef5f15c388
